src/tokenizer.java

```diff
@@ -21,152 +21,115 @@
-
-                    i++;
-                }
-
-                //Handle the gathered token
-                if ("=".equals(tok.toString())) {
-                    tokenList.add(new asmt_op(lineCount));
-                }
-                else if ("charAt".equals(tok.toString())) {
-                    tokenList.add(new charAt_label(lineCount));
-                }
-                else if (",".equals(tok.toString())) {
-                    tokenList.add(new comma(lineCount));
-                }
-                else if ("concat".equals(tok.toString())) {
-                    tokenList.add(new concat_label(lineCount));
-                }
-                else if ("Double".equals(tok.toString())) {
-                    tokenList.add(new double_label(lineCount));
-                }
-                else if (")".equals(tok.toString())) {
-                    tokenList.add(new end_paren(lineCount));
-                }
-                else if (";".equals(tok.toString())) {
-                    tokenList.add(new end_stmt(lineCount));
-                }
-                else if ("Integer".equals(tok.toString())) {
-                    tokenList.add(new int_label(lineCount));
-                }
-                else if ("print".equals(tok.toString())) {
-                    tokenList.add(new print_label(lineCount));
-                }
-                else if ("(".equals(tok.toString())) {
-                    tokenList.add(new start_paren(lineCount));
-                }
-                else if ("String".equals(tok.toString())) {
-                    tokenList.add(new str_label(lineCount));
-                }
-                else if ("*".equals(tok.toString()) || "/".equals(tok.toString()) || "^".equals(tok.toString()) ||
-                        "+".equals(tok.toString()) || "-".equals(tok.toString())) {
-                    tokenList.add(new op(lineCount, tok.toString()));
-                }
-                else if ("\"".equals(tok.toString())) {
-                    //String literal
-                    tokenList.add(new quote(lineCount));
-
-                    tok = new StringBuilder();
-
                     i++;
                     while (i < lineLength) {
-                        char ch = line.charAt(i);
-                        if (ch == '"')
+                        ch = line.charAt(i);
+                        if (ch == '"') {
+                            tok.append(ch);
                             break;
-
-                        if (!((ch >= 'A' && ch <= 'Z') || (ch >= 'a' && ch <= 'z') || (ch >= '0' && ch <= '9') || ch == ' '))
+                        }
+                        if (!Character.isAlphabetic(ch) && !Character.isDigit(ch) && ch != ' ')
                             errorPrinter.throwError(lineCount, new Syntax("Illegal character '" + ch +
                                     "' detected in string literal"));
-
                         tok.append(ch);
-
                         i++;
                     }
```

Comment:
Just realized that the changes on tokenizer branch is reflected on here; Shifting all comments onto this PR - 

Just want to eliminate that double `i++`, remove the `i++` before while loop and at end of while loop:

```suggestion
                   while (i < lineLength) {
                        i++;
                        char ch = line.charAt(i);
```

Reply:
The `i++` before the while loop handles a str_literal that starts at the end of the line and immediately wraps around. In the suggestion, `line.charAt(i)` may throw an IndexOutOfBounds. Im sorry for the lack of documentation.